src/services/identity/UsersService.ts

```diff
@@ -189,20 +189,21 @@ class UsersService {
   async sendEmailOtp(email: string) {
     const parsedEmail = email.toLowerCase()
     const { otp, hashedOtp } = await this.otpService.generateLoginOtpWithHash()
+    const data = {
+      hashedOtp,
+      attempts: 0,
+      expiresAt: this.getOtpExpiry(),
+    }
 
     // Reset attempts to login
-    const otpEntry = await this.otpRepository.findOne({
+    // Reset attempts to login
+    const [otpEntry, created] = await this.otpRepository.findOrCreate({
       where: { email: parsedEmail },
+      defaults: data,
     })
-    if (!otpEntry) {
-      // create new entry
-      await this.createOtpEntry(parsedEmail, OtpType.Email, hashedOtp)
-    } else {
-      await otpEntry?.update({
-        hashedOtp,
-        attempts: 0,
-        expiresAt: this.getOtpExpiry(),
-      })
+
+    if (!created) {
+      await otpEntry?.update(data)
```

Comment:
nit: actually because the data is already available prior and the important thing here is just to create, maybe `upsert` might be a better fit?

Reply:
You're right! `upsert` is the pattern being used in both the refactor PRT I opened, I'll do one more round of edits

Reply:
Done https://github.com/isomerpages/isomercms-backend/pull/1283/commits/1a15704fa4832d49b3cc44973697bf2536f1d751